Build a regression model.

In [ ]:
import pandas as pd
import statsmodels.api as sm


merged_df = pd.read_csv('/home/t0si/Statistical_Modelling_Project/data/merged_data.csv')


data = merged_df[['rating', 'distance', 'price', 'review_count', 'free_bikes']].dropna()
data = data[(data['price'] > 0) & (data['review_count'] > 0)]


X = data[['distance', 'price', 'review_count', 'free_bikes']]
y = data['rating']


X = sm.add_constant(X)


model = sm.OLS(y, X)
results = model.fit()


print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                 rating   R-squared:                       0.021
Model:                            OLS   Adj. R-squared:                  0.017
Method:                 Least Squares   F-statistic:                     5.086
Date:                Mon, 18 Nov 2024   Prob (F-statistic):           0.000468
Time:                        21:18:00   Log-Likelihood:                -799.01
No. Observations:                 935   AIC:                             1608.
Df Residuals:                     930   BIC:                             1632.
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const            3.8512      0.068     56.812   

Provide model output and an interpretation of the results. 

The model has a very low R², explaining only 2.1% of the variance in rating. This indicates the predictors are not strong drivers of the dependent variable.
The model shows that the factors studied have little impact on ratings, with only distance having a small, likely insignificant effect, and the results may not be fully reliable.




# Stretch

How can you turn the regression model into a classification model?

To turn the regression model into a classification model, I would need to transform the rating data and divide the ratings into categories. One option would be to classify them as low, medium, or high. This approach would enable the classification model to predict categories instead of the actual ratings.